### ISSUE TITLE
crypto.createCredentials is deprecated

### ISSUE THREAD

**Issue by chpio** [node-xmpp/node-xmpp-core#82]
Hi,

im getting the warning: `(node) crypto.createCredentials is deprecated. Use tls.createSecureContext instead.`.

i think it's coming from the *tls-connect* module

**Comment by zacharyabresch** [comment]
+1

**Comment by chpio** [comment]
uhh, sorry for double

**Comment by zacharyabresch** [comment]
Double? Is there another issue about this? I'm banging my head against a wall! :)

**Comment by chpio** [comment]
ohh, thought my issue was a double to your issue...

**Comment by zacharyabresch** [comment]
No. I'm just having the same issue. I got it to go away at some point but after some updates it came back. I thought it was originally an issue with [node-simple-xmpp](https://github.com/simple-xmpp/node-simple-xmpp) but it turned out to be an issue in node-xmpp-core. See my issue [here](https://github.com/simple-xmpp/node-simple-xmpp/issues/66). I've reverted my `node` version back and am still having the issue.

**Comment by zacharyabresch** [comment]
I'm using v3.0.0 of `node-xmpp-client`.

**Comment by zacharyabresch** [comment]
Umm ... never mind all that. It turns out, my QA team pushed an update to our server that broke authentication. :-1:

**Comment by andrewaustin** [comment]
This is caused by tls-connect, which looks to be a dead package. See https://github.com/dodo/node-tls-connect/issues/2 Any easy way to fix would be to link against ```github.com/fritx/node-tls-connect/tree/fix-tls-deprecate``` instead of ```^0.2.2``` in package.json. The other option is to remove the dependency on tls-connect since it shouldn't be needed in recent versions of node.

**Comment by drakoumel** [comment]
This warning comes from

    at Connection.setSecure (/Users/chriszioutas/Development/bmo/node_modules/node-xmpp-client/node_modules/node-xmpp-core/lib/Connection.js:320:19)

So it is in node-xmpp-core.

**Comment by chpio** [comment]
@drakoumel 
o.0

```javascript
var starttls = require('tls-connect')
// ...
// line 320:
var cleartext = starttls({
// ....
```

https://github.com/node-xmpp/node-xmpp-core/blob/master/lib/Connection.js#L320

**Comment by sonnyp** [comment]
https://github.com/node-xmpp/node-xmpp-core/pull/85

**Comment by sonnyp** [comment]
fixed in 5.0.1
